engine/src/main/java/org/terasology/identity/storageServiceClient/StorageServiceWorker.java

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright 2017 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.identity.storageServiceClient;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Maps;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.terasology.config.Config;
+import org.terasology.config.IdentityStorageServiceConfig;
+import org.terasology.config.SecurityConfig;
+import org.terasology.context.Context;
+import org.terasology.i18n.TranslationSystem;
+import org.terasology.identity.ClientIdentity;
+import org.terasology.identity.PublicIdentityCertificate;
+import org.terasology.logic.console.Console;
+import org.terasology.logic.console.CoreMessageType;
+
+import java.net.URL;
+import java.util.Stack;
+import java.util.Map;
+
+/**
+ * The public interface to this package. Manages a communication session with the storage service server,
+ * can answer for status information queries and can perform asynchronous operations on the storage service.
+ * This class can be in two states, with the default being "logged out".
+ */
+public final class StorageServiceWorker {
+
+    private static final Logger logger = LoggerFactory.getLogger(StorageServiceWorker.class);
+
+    StorageServiceWorkerStatus status = StorageServiceWorkerStatus.LOGGED_OUT;
+    APISession sessionInstance;
+    String loginName;
+    Stack<IdentityBundle> conflictingRemoteIdentities;
+
+    final IdentityStorageServiceConfig storageConfig;
```

Comment:
Yes it's intended to be shared (like the other non-private fields `status`, `sessionInstance`, etc). Basically, all the `Action`s access the `StorageServiceWorker` by accessing and sometimes modifying it's fields; by using the default access modifier, this is only possible from the inside of the package, so it shouldn't be possible to bring the class to an inconsistent state from the outside.
I'm sorry if this doesn't look like a very good practice, but using the `Action`s is the only way I found to avoid making a huge `StorageServiceWorker` class. I didn't use getters and setters since they wouldn't make a big difference (each of these fields would be class-private and have a package-private getter and setter). I'm open to any suggestions to improve this.